app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,79 @@
+class TweetsController < ApplicationController
+  get '/tweets' do
+  if logged_in?
+    @tweets = Tweet.all
+ erb :'tweets/tweets'
+else
+  redirect to '/login'
+  end
+end
+
+  get '/tweets/new' do
+  if logged_in?
+  erb :'/tweets/create_tweet'
+else
+  redirect to '/login'
+end
+end
+
+post '/tweets' do
+   if params[:content] == ""
+     redirect to "/tweets/new"
+   else
+     current_user.tweets << @tweet = Tweet.create(content: params[:content])
+    redirect to "/tweets/#{@tweet.id}"
+   end
+ end
+
+
+  get '/tweets/:id' do
+    if logged_in?
+    @tweet = Tweet.find_by_id(params[:id])
+  erb :'tweets/show_tweet'
+else
+  redirect to '/login'
+  end
+end
+
+
+get '/tweets/:id/edit' do
+  if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == current_user.id
+       erb :'tweets/edit_tweet'
+      else
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+
+
+  patch '/tweets/:id' do
+    if params[:content] == ""
+        redirect to "/tweets/#{params[:id]}/edit"
+    else
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...